File: app/src/main/java/com/originb/inkwisenote2/modules/noterelation/data/TextProcessingStage.java
package com.originb.inkwisenote2.modules.noterelation.data;

public enum TextProcessingStage {
    TEXT_PARSING("text_parsing"),
    TOKENIZATION("tokenization"),
    NOTE_READY("note_ready");

    private final String textProcessingStage;

    TextProcessingStage(String textProcessingStage) {
        this.textProcessingStage = textProcessingStage;
    }

    @Override
    public String toString() {
        return textProcessingStage;
    }

    public boolean isEqualTo(String stage) {
        return this.textProcessingStage.equals(stage);
    }
}

File: app/src/main/java/com/originb/inkwisenote2/modules/noterelation/data/TextProcessingStage.kt
package com.originb.inkwisenote2.modules.noterelation.data

enum class TextProcessingStage(private val textProcessingStage: String) {
    TEXT_PARSING("text_parsing"),
    TOKENIZATION("tokenization"),
    NOTE_READY("note_ready");

    override fun toString(): String {
        return textProcessingStage
    }

    fun isEqualTo(stage: String): Boolean {
        return this.textProcessingStage == stage
    }
}